src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,200 @@
 package core.basesyntax;
 
+import java.util.Iterator;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> last = tail;
+        Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new IndexOutOfBoundsException("Incorrect index " + index);
        }
        if (index == size) {
            add(value);
            return;
        }
```

Reply:
I don't agree with you, as we should first check if we get the correct index and then call add method. And checkIndexBounds() it's quite another condition, as we should let in this case (index == size).

Reply:
Artem said a good improvement. That decreases your code. You have almost the same now, but with additional almost duplicated code. Look, if `index == size` - you just call the method that adds a new node to the end of the list, otherwise - you check an index using the method `checkIndexBounds(index);` which contains almost the same block of code: `index < 0 || index >= size`.